### PR TITLE
Adds a kneecapping element and adds that very element to baseball bats.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

image
image

Implements appropriate signal handlers for `/obj/item/proc/attack_secondary()` - Secondary item attacks now send signals and signal handlers can now cancel the attack chain, continue it or move to the primary attack chain.

Adds a kneecapping element.

Kneecapping element replaces the item's secondary attack with an aimed attack at the kneecaps under certain circumstances. If these circumstances are not met, the secondary attack chain will continue as normal with the item's normal secondary attack.

Element is incompatible with non-items. Requires the parent item to have a force equal to or greater than WOUND_MINIMUM_DAMAGE. Also requires that the parent can actually get past pre_secondary_attack without the attack chain cancelling. Funnily enough, the Element doesn't require a blunt weapon. If admins add it to sharp or pointy weapons, they get the same aimed attack but with sharp and pointy wounds instead - Because all the Element does is add a hefty wounding modifier.

Kneecapping attacks have a wounding bonus between severe and critical+10 wound thresholds. Without some serious wound protecting armour this all but guarantees a wound of some sort. The attack is directed specifically at a limb and the limb takes the damage.

Requires the attacker to be aiming for either leg zone, which will be targetted specifically. They will than have a 3-second do_mob before executing the attack.

Kneecapping requires the target to either be on the floor, immobilised or buckled to something.

Passing all the checks will cancel the entire attack chain for the Element to handle attacking logic.

This Element has been added to the baseball bat. Any other items that wish to implement its functionality in the future are free to.

## Why It's Good For The Game

Now you can genuinely break people's kneecaps, causing near guaranteed wounds to the legs.

Another secondary attack added, leveraging the awesome power of right clicking things.

A lot of RP potential. Become a loan shark and break the kneecaps of defaulters. Apply inhanced interrogation to your suspects. Show that asshole monkey who's boss.

Gives medbay more work.

## Changelog
:cl:
add: Baseball bats can now be used to break kneecaps with their right click attack. This requires the target to be one of either prone, immobilised or buckled to something. If you aim for the legs and right click, you'll wind up a swing for the kneecaps. Research indicates that this isn't good for the target's legs.
/:cl:

### CODE REVIEW

**Comment by KathyRyals** [comment]
Amazing, wow.

**Comment by ATH1909** [comment]
an extra way to fuck up the corpses of your victims, nice

File: code/datums/components/kneecapping.dm [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	attacker.visible_message(span_warning("[attacker] carefully aims [attacker.p_their()] [weapon] for a swing at [target]'s kneecaps!"), span_danger("You carefully aim \the [weapon] for a swing at [target]'s kneecaps!"))
+	log_combat(attacker, target, "started aiming a swing to break the kneecaps of", weapon)
+
+	if(do_mob(attacker, target, 3 SECONDS, interaction_key = weapon))
+		attacker.visible_message(span_warning("[attacker] swings [attacker.p_their()] [weapon] at [target]'s kneecaps!"), span_danger("You swing \the [weapon] at [target]'s kneecaps!"))
+		var/datum/wound/blunt/severe/severe_wound_type = /datum/wound/blunt/severe
+		var/datum/wound/blunt/critical/critical_wound_type = /datum/wound/blunt/critical
+		leg.receive_damage(brute = weapon.force, wound_bonus = rand(initial(severe_wound_type.threshold_minimum), initial(critical_wound_type.threshold_minimum) + 10))
+		log_combat(attacker, target, "broke ther kneecaps of", weapon)
```
Comment (Fikou) [inline code/datums/components/kneecapping.dm:87]: ther kneecaps

File: code/datums/components/kneecapping.dm [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	if(LAZYACCESS(attacker.do_afters, weapon))
+		return
+
+	attacker.visible_message(span_warning("[attacker] carefully aims [attacker.p_their()] [weapon] for a swing at [target]'s kneecaps!"), span_danger("You carefully aim \the [weapon] for a swing at [target]'s kneecaps!"))
+	log_combat(attacker, target, "started aiming a swing to break the kneecaps of", weapon)
+
+	if(do_mob(attacker, target, 3 SECONDS, interaction_key = weapon))
+		attacker.visible_message(span_warning("[attacker] swings [attacker.p_their()] [weapon] at [target]'s kneecaps!"), span_danger("You swing \the [weapon] at [target]'s kneecaps!"))
+		var/datum/wound/blunt/severe/severe_wound_type = /datum/wound/blunt/severe
+		var/datum/wound/blunt/critical/critical_wound_type = /datum/wound/blunt/critical
```
Comment (ATH1909) [inline code/datums/components/kneecapping.dm:85]: what if this element has been applied to a non-blunt weapon?

Comment (Fikou) [reply code/datums/components/kneecapping.dm:85]: well fuck the person that did it then

**Comment by Fikou** [comment]
i feel like this interaction is too fast for giving someone a huge slowdown and making their limb really easy to remove, something like the amputation shears is 15 seconds

File: code/datums/components/kneecapping.dm (hunk 1 above)
Comment (ATH1909) [reply code/datums/components/kneecapping.dm:87]: them thar kneecaps

**Comment by ATH1909** [comment]
> i feel like this interaction is too fast for giving someone a huge slowdown and making their limb really easy to remove, something like the amputation shears is 15 seconds

it takes the same amount of time to do this as it takes to cuff someone, which is far greater of a "you lose" effect

**Comment by MrDoomBringer** [comment]
amazed this wasn't a shaps PR good work

**Comment by Mickyan** [comment]
Can't baseball bats still stunlock people indefinitely

File: code/datums/components/kneecapping.dm (hunk 1 above)
Comment (Timberpoes) [reply code/datums/components/kneecapping.dm:87]: I haff marburrs in my mouf. Ther is no way I wud maek such a mistake.

Comment (Timberpoes) [inline code/datums/components/kneecapping.dm:87]:
```suggestion
		log_combat(attacker, target, "broke the kneecaps of", weapon)
```

Comment (Timberpoes) [reply code/datums/components/kneecapping.dm:87]: We call this the Dennok edit.

File: code/datums/components/kneecapping.dm (hunk 2 above)
Comment (Timberpoes) [reply code/datums/components/kneecapping.dm:85]: I just picked blunt as the baseline for wounding interactions out of my ass.

Even though the element can technically be added to any item with enough force, from var edited plastic spoons to spears, the behaviour can be further enhanced later if someone wanted to add checks for the injury thresholds for piercing and slashing wounds too.

Right now the element is only added to a single blunt weapon outside of admin abuse. I wanted to keep the option open for admins to add it to any weapon even if it doesn't use equivalent threasholds for that weapon's wounding type.

**Pull request merged**